### PR TITLE
Adapt limit to query db view as per couch db change

### PR DESCRIPTION
**Pull request by SugandhaAgrawal**
Co-authored-by: steffenrost <srost@de.ibm.com>

CouchDB has introduced a limit check for querying the database view, for the query parameter `limit`. In OpenWhisk it was first set to Scala `Int.Max` value. Since this exceeds the allowed range by couchDB, the query calls using the max value have been reported failing with `http Response: 400 Bad Request`. 

For ref: https://github.com/apache/couchdb/commit/1fdfe466223380d1ee7c82e21283e161290aed87#diff-2bd31f00dcfba780c6c65bc69c256854 

## Description
Changed the query parameter `limit` to max allowed by couchDB. 

## Related issue and scope

- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [x] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes

- [x] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: core/invoker/src/main/scala/org/apache/openwhisk/core/invoker/NamespaceBlacklist.scala [hunk 1]
```diff
@@ -38,6 +38,9 @@ class NamespaceBlacklist(authStore: AuthStore) {
 
   private var blacklist: Set[String] = Set.empty
 
+  // Enforced by couchdb, max limit value for db view query
+  private var couchdb_limit: Int = 268435456
```
Comment (markusthoemmes) [inline core/invoker/src/main/scala/org/apache/openwhisk/core/invoker/NamespaceBlacklist.scala]: This should be a `val` and as it's a constant should be written in all caps, like `COUCHDB_LIMIT`.

However, I feel like this should be factored into the couchdb code itself and the `limit` value should have an upper-bound applied there. Technically you do not know in this piece of code that you're handling CouchDB.

Comment (SugandhaAgrawal) [reply core/invoker/src/main/scala/org/apache/openwhisk/core/invoker/NamespaceBlacklist.scala]: Fair point!

**Comment by rabbah** [comment]
> In OpenWhisk it was first set to Scala Int.Max value.

Can you point me to it? I thought it was set to `0` not max int.

**Comment by rabbah** [comment]
OK I found this reference https://github.com/apache/incubator-openwhisk/blob/be1e3a19c02956c9be85023a0bb0ff399c21444d/core/invoker/src/main/scala/org/apache/openwhisk/core/invoker/NamespaceBlacklist.scala#L57

This should just be set to 0 which is what the collections do.

**Comment by steffenrost** [comment]
> See my comments - the simpler change is to change Int.Max to 0.

But setting the limit query parameter to 0 (`limit = 0,`)would limit the number of returned blocked namespace in the result set to 0, is that the intended behavior?

**Comment by rabbah** [comment]
The couchdb client in openwhisk will omit the `limit` query parameter when the value is set to `0`:
https://github.com/apache/incubator-openwhisk/blob/be1e3a19c02956c9be85023a0bb0ff399c21444d/common/scala/src/main/scala/org/apache/openwhisk/core/database/CouchDbRestClient.scala#L124

So the caller (here the invoker) should set the value to `0` to indicate no limit.

**Comment by rabbah** [comment]
In general, using "no limit" seems wrong here anyway and the invoker should paginate since it's conceivable there could be a large enough response to both cause a performance problem and a memory problem. In practice, I presume the responses are typically small not to be an issue.

**Comment by SugandhaAgrawal** [comment]
@rabbah - Adapted as our discussion. Please Review!!! TIA

File: core/invoker/src/main/scala/org/apache/openwhisk/core/invoker/NamespaceBlacklist.scala [hunk 2]
```diff
@@ -47,14 +47,15 @@ class NamespaceBlacklist(authStore: AuthStore) {
   def isBlacklisted(identity: Identity): Boolean = blacklist.contains(identity.namespace.name.asString)
 
   /** Refreshes the current blacklist from the database. */
+  /** Limit query parameter set to 0 for limitless record query */
```
Comment (rabbah) [inline core/invoker/src/main/scala/org/apache/openwhisk/core/invoker/NamespaceBlacklist.scala]: Please terminate the sentence with a period.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/OpenWhisk-Playground/86/
Test FAILed.
Test FAILured.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/OpenWhisk-Playground/87/
Test FAILed.
Test FAILured.

**Pull request merged**